lib/Controller/ImportController.php

```diff
@@ -48,33 +48,81 @@ public function __construct(
 		$this->l10n = $l10n;
 	}
 
+	/**
+	 * @NoAdminRequired
+	 */
+	public function previewImportTable(int $tableId, String $path): DataResponse {
+		return $this->handleError(function () use ($tableId, $path) {
+			return $this->service->previewImport($tableId, null, $path);
+		});
+	}
+
+	/**
+	 * @NoAdminRequired
+	 */
+	public function importInTable(int $tableId, String $path, bool $createMissingColumns = true, array $columnsConfig = []): DataResponse {
+		return $this->handleError(function () use ($tableId, $path, $createMissingColumns, $columnsConfig) {
+			return $this->service->import($tableId, null, $path, $createMissingColumns, $columnsConfig);
+		});
+	}
 
 	/**
 	 * @NoAdminRequired
 	 */
-	public function importInTable(int $tableId, String $path, bool $createMissingColumns = true): DataResponse {
-		return $this->handleError(function () use ($tableId, $path, $createMissingColumns) {
-			return $this->service->import($tableId, null, $path, $createMissingColumns);
+	public function previewImportView(int $viewId, String $path): DataResponse {
+		return $this->handleError(function () use ($viewId, $path) {
+			return $this->service->previewImport(null, $viewId, $path);
 		});
 	}
 
 	/**
 	 * @NoAdminRequired
 	 */
-	public function importInView(int $viewId, String $path, bool $createMissingColumns = true): DataResponse {
-		return $this->handleError(function () use ($viewId, $path, $createMissingColumns) {
-			return $this->service->import(null, $viewId, $path, $createMissingColumns);
+	public function importInView(int $viewId, String $path, bool $createMissingColumns = true, array $columnsConfig = []): DataResponse {
+		return $this->handleError(function () use ($viewId, $path, $createMissingColumns, $columnsConfig) {
+			return $this->service->import(null, $viewId, $path, $createMissingColumns, $columnsConfig);
 		});
 	}
 
 	/**
 	 * @NoAdminRequired
 	 */
-	public function importUploadInTable(int $tableId, bool $createMissingColumns = true): DataResponse {
+	public function previewUploadImportTable(int $tableId): DataResponse {
+		try {
+			$file = $this->getUploadedFile('uploadfile');
+			return $this->handleError(function () use ($tableId, $file) {
+				return $this->service->previewImport($tableId, null, $file['tmp_name']);
+			});
+		} catch (UploadException | NotPermittedException $e) {
+			$this->logger->error('Upload error', ['exception' => $e]);
+			return new DataResponse(['message' => $e->getMessage()], Http::STATUS_BAD_REQUEST);
+		}
+	}
+
+	/**
+	 * @NoAdminRequired
+	 */
+	public function importUploadInTable(int $tableId, bool $createMissingColumns = true, string $columnsConfig = ''): DataResponse {
```

Comment:
Why are we passing a string here and json_decode while all other methods use an array?

Reply:
Ah becaues of the file upload with multipart data, then seems fine 👍